src/path/fs.rs

```diff
@@ -70,6 +73,20 @@ impl Predicate<path::Path> for BinaryFilePredicate {
     }
 }
 
+impl Predicate<str> for BinaryFilePredicate {
```

Comment:
Why is this implementation being added?

Reply:
@epage @kammitama5 and I had a look at this. You can't compare without this implementation. The example code will call `Predicate<I: Item>::call(BinaryFilePredicate, I)` so without this implementation you'll get the following errors.

```
error[E0277]: the trait bound `predicates::path::BinaryFilePredicate: predicates::Predicate<str>` is not satisfied
 --> src/path/fs.rs:110:34
  |
9 | assert_eq!(false, predicate_file.eval("src"));
  |                                  ^^^^ the trait `predicates::Predicate<str>` is not implemented for `predicates::path::BinaryFilePredicate`
  |
  = help: the following implementations were found:
            <predicates::path::BinaryFilePredicate as predicates::Predicate<std::path::Path>>
            <predicates::path::BinaryFilePredicate as predicates::Predicate<[u8]>>
```

Reply:
So this is being used to try to emulate `AsRef<Path>` with `eval`?

The original issue was only about `eq_file`.  In thinking about the trade offs of adding this extra functionality, I'm leaning against it.  Without this, the type system enforces what categories of information a predicate can deal with.  This change removes that and makes it easy to use a `eq_file` predicate with any string predicate situation causing problems.

It might seem nice to users but I am expecting the main use case of predicates is that users create them and other libraries (like assert_cmd) consume them, so improving usability for this side doesn't offer as much benefit.

Reply:
@epage I don't see why that use case would justify a more unintuitive API, I think assumes the people writing the predicates have a pre existing level of familiarity with predicates, which will never always be the case. Since these predicates are mostly going to be used in tests, most of the time the paths are going to be str literals anyway, why add the extra noise of `Path::new("…")` to their tests?

Reply:
As I said, my expectation is that the call to `eval` is not in the test but in a crate the test calls.  In that case, string literals wouldn't be used and the crate author is probably dealing with a specific type anyways.

I also just checked the implementation.  This conflicts with the intention of how the other predicate impls work:

```rust
impl Predicate<Path> for BinaryFilePredicate
impl Predicate<[u8]> for BinaryFilePredicate

impl Predicate<Path> for StrFilePredicate
impl Predicate<str> for StrFilePredicate
```

The non-path versions of the predicates allow using `eq_file` with in-memory test-values.  Adding `impl Predicate<str> for BinaryFilePredicate` then has a different meaning then `impl Predicate<[u8]> for BinaryFilePredicate` and cannot be added to `StrFilePredicate`

Reply:
@epage I can agree that with that inconsistency would make it less than ideal, and I can't really find a better alternative.So I'm fine with that being removed.

Reply:
I will be sure to reference the issue and squash when given the ok. Thank you so much, both of you! :)